apps/files_sharing/src/views/SharingDetailsTab.vue

```diff
@@ -0,0 +1,525 @@
+				</NcButton>
+				<NcButton type="primary" @click="shareAndCopyLink">
+					{{ t('file_sharing', 'Share & copy link') }}
+				</NcButton>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcInputField from '@nextcloud/vue/dist/Components/NcInputField.js'
+import NcDatetimePicker from '@nextcloud/vue/dist/Components/NcDatetimePicker.js'
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+import ShareIcon from 'vue-material-design-icons/ShareVariant.vue'
+import CloseIcon from 'vue-material-design-icons/Close.vue'
+import EditIcon from 'vue-material-design-icons/Pencil.vue'
+import ViewIcon from 'vue-material-design-icons/Eye.vue'
+import UploadIcon from 'vue-material-design-icons/Upload.vue'
+import MenuDownIcon from 'vue-material-design-icons/MenuDown.vue'
+import DotsHorizontalIcon from 'vue-material-design-icons/DotsHorizontal.vue'
+
+/// Temp
+
+import NcActions from '@nextcloud/vue/dist/Components/NcActions.js'
+import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton.js'
+import NcActionCheckbox from '@nextcloud/vue/dist/Components/NcActionCheckbox.js'
+import NcActionInput from '@nextcloud/vue/dist/Components/NcActionInput.js'
+import NcActionTextEditable from '@nextcloud/vue/dist/Components/NcActionTextEditable.js'
+
+/// End temp
+
+/// Migrated from ShareInput.vue
+
+import Config from '../services/ConfigService.js'
+import GeneratePassword from '../utils/GeneratePassword.js'
+import Share from '../models/Share.js'
+import ShareRequests from '../mixins/ShareRequests.js'
+import ShareTypes from '../mixins/ShareTypes.js'
+
+/// Migrated from SharingEntry
+
+import SharesMixin from '../mixins/SharesMixin.js'
+
+export default {
+	name: 'SharingDetailsTab',
+	components: {
+		NcButton,
+		NcInputField,
+		NcDatetimePicker,
+		NcCheckboxRadioSwitch,
+		ShareIcon,
+		CloseIcon,
+		EditIcon,
+		ViewIcon,
+		UploadIcon,
+		MenuDownIcon,
+		DotsHorizontalIcon,
+		NcActions, // To be removed
+		NcActionButton, // To be removed
+		NcActionCheckbox, // To be removed
+		NcActionInput, // To be removed
+		NcActionTextEditable, // To be removed
+	},
+	mixins: [ShareTypes, ShareRequests],
+	props: {
+		shareRequestValue: {
+			type: Object,
+			required: false,
+		},
+		fileInfo: {
+			type: Object,
+			required: true,
+		},
+		share: {
+			type: Object,
+			default: false,
+		},
+	},
+	data() {
+		return {
+			setPassword: false,
+			password: '',
+			noteToRecipient: '',
+			sharingPermission: 'rw',
+			hasExpirationDate: false,
+			expirationDate: '',
+			hideDownload: false,
+			shareLabel: '',
+			setCustomPermissions: false,
+			advancedSectionAccordionExpanded: false,
+			permissionsEdit: false,
+			permissionsCreate: false,
+			permissionsDelete: false,
+			permissionsRead: false,
+			permissionsShare: false,
+		}
+	},
+
+	computed: {
+		title() {
+			let title = 'Share with '
```

Comment:
Should be translated, ideally also with the shareWithDisplayName parameter included, so that translators have context. This should work, but we may nee to think about a fallback for types that don't have a shareWithDisplayName

```suggestion
			let title = t('Share with {displayName}', { displayName: this.share.shareWithDisplayName })
```